### DISCUSSION TITLE
Python 3.6 syntax error line 184/185 in settings.py?

### DISCUSSION THREAD

**Discussion by 4fd81048-Brian** [netbox-community/netbox#6673] [Q&A] [answered]
Maybe this is because I'm using Python 3.6, on Centos 7.9, but settings.py line 184 and 185 are causing me a syntax error with manage.py nbshell.  If I remove the "f" it gets past that (but fails at from urllib.parse import urlsplit ImportError: No module named parse which is a different issue)

```
[me@clipper netbox]$ sudo netbox/manage.py nbshell
Traceback (most recent call last):
  File "netbox/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/lib/python2.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/usr/lib/python2.7/site-packages/django/core/management/__init__.py", line 308, in execute
    settings.INSTALLED_APPS
  File "/usr/lib/python2.7/site-packages/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/usr/lib/python2.7/site-packages/django/conf/__init__.py", line 41, in _setup
    self._wrapped = Settings(settings_module)
  File "/usr/lib/python2.7/site-packages/django/conf/__init__.py", line 110, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/lib64/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/opt/netbox/netbox/netbox/settings.py", line 184
    f"STORAGE_BACKEND is set to {STORAGE_BACKEND} but django-storages is not present. It can be"
                                                                                               ^
SyntaxError: invalid syntax
```

**Comment by candlerb** [comment, marked as answer]
If you look at your backtrace, rather clearly you are running this under python 2.7 - and hence the error.

To run nbshell, you need to activate your virtualenv first:

```
cd /opt/netbox
. venv/bin/activate
python3 netbox/manage.py nbshell
```

**Reply by 4fd81048-Brian** [reply]
Thank-you!  I just noticed that as well and it had me scratching my head.

It works the way you wrote, and I had also edited the first line of manage.py to #!/usr/bin/env python3 (adding the 3) and that worked too.  I think I'm fighting a losing battle with this upgrade on Centos.

**Reply by candlerb** [reply]
> I had also edited the first line of manage.py to #!/usr/bin/env python3 (adding the 3) and that worked too.

That's not a good idea.  You always want to run things inside the venv - otherwise you'll end up using old, stale copies of packages installed globally on your system.

Upgrade should be straightforward: it basically boils down to running `./upgrade.sh` (which *doesn't* need to be inside the venv, as it builds the venv for you).

If you are having problems, then we may be able to help if you give more specific info:
- what version of Netbox you're upgrading from and to
- whether Netbox was installed from a tarball or a git checkout (does directory `/opt/netbox/.git` exist?)
- the commands you've typed and the responses you see (exact copy-paste)
- anything else relevant, e.g. you're trying to change from supervisord to systemd
